recipes/libcoap/all/conanfile.py

```diff
@@ -0,0 +1,100 @@
+import os
+from conans import CMake, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class LibCoapConan(ConanFile):
+    name = "libcoap"
+    license = "BSD-2-Clause"
+    homepage = "https://github.com/gocarlos/libcoap/"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = """A CoAP (RFC 7252) implementation in C"""
+    topics = ("coap")
+    exports_sources = "CMakeLists.txt"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_epoll": [True, False],
+        "dtls_backend": [None, "openssl", "gnutls", "tinydtls", "mbedtls"],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_epoll": False,
+        "dtls_backend": "openssl",
+    }
+    generators = "cmake", "cmake_find_package"
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def requirements(self):
+        if self.options.dtls_backend == "openssl":
+            self.requires.add("openssl/1.1.1g")
+        if self.options.dtls_backend == "mbedtls":
+            self.requires.add("mbedtls/2.16.3-apache")
+        if self.options.dtls_backend == "gnutls":
+            raise ConanInvalidConfiguration("gnu tls not available yet")
+        if self.options.dtls_backend == "tinydtls":
+            raise ConanInvalidConfiguration("tinydtls not available yet")
+
+    def _patch_files(self):
```

Comment:
```suggestion
    def _patch_files(self):
        # TODO: Remove custom targets when Conan components be available.
```

Just a small to help in the future. So we can filter running a grep -r